proto/lighthouse-result.proto

```diff
@@ -139,6 +139,9 @@ message LighthouseResult {
     // List of the categories that were run, empty if all were run
     // nullable list of strings
     google.protobuf.Value only_categories = 3;
+
+    // How Lighthouse was run, e.g. from the Chrome extension or from the npm module
+    string channel = 4;
```

Comment:
If we are adding it to the proto, I would do it as an enum, i.e.
```
    enum Channel {
      UNKNOWN_CHANNEL = 0;
      ext = 1;
      cdt = 2;
      cli = 3;
      nm = 4;
      lr = 5;
    }
    Channel channel = 4;
```

But I would confirm with @paulirish we would want to add it.  We have elided much of the configSettings from the proto.  Esp. since the channel would always be `lr` for proto LHRs I'm not 100% it would be useful 🤷‍♂️

Reply:
Ok, if we don't include it then when rendering the report we don't know it was generated by Lightrider? The channel value would just be `undefined`?

@rviscomi Do we care much about being able to identify Lightrider-generated reports from the UTM links?

Reply:
Oh yeah.  If it is an LR generated LHR and the proto didn't include this new field the `channel` field would be totally elided.  If we need this info from a rendering side, we would need to add this new field.  Which to me is a valid justification to include it in the proto imo.  

If we use the enum we need to include `UNKNOWN_CHANNEL` in the js so that can propagate from protos as a valid value that people would need to expect.

Reply:
It'd be nice to have complete coverage of all channels but not critical.

Reply:
@exterkamp Could PSI and web.dev pass in their own channel values and override the Lightrider default?

Reply:
PSI could definitely pass it's own channel.  Web.dev, we'll have to make sure.  @kusler?